docs/rest.md

```diff
@@ -0,0 +1,623 @@
+}
+```
+
+## /1.0/packages/[name]/services/[name]/logs
+
+### GET
+
+* Description: Logs for the service from a package
+* Authorization: trusted
+* Operation: sync
+* Return: service logs
+
+#### Sample result:
+
+```javascript
+[
+   {
+       "timestamp": "1440679470679901",
+       "message": "something happened",
+       "raw": {}
+   },
+   {
+       "timestamp": "1440679470680968",
+       "message": "bla bla",
+       "raw": {}
+    }
+]
+```
+
+## /1.0/packages/[name]/config
+
+Query an active package for information about its configuration, and alter
+that configuration. Will return an error if the package is not active.
+
+### GET
+
+* Description: Configuration for a package
+* Authorization: trusted
+* Operation: sync
+* Return: package configuration
+
+#### Sample result:
+
+```javascript
+"config:\n  ubuntu-core:\n    autopilot: false\n    timezone: Europe/Berlin\n    hostname: localhost.localdomain\n"
+```
+
+Notes: user facing implementations in text form must show this data using yaml.
+
+### PUT
+
+* Description: Set configuration for a package
+* Authorization: trusted
+* Operation: sync
+* Return: package configuration
+
+#### Sample input:
+
+```javascript
+        config:\n  ubuntu-core:\n    autopilot: true\n
+```
+
+#### Sample result:
+
+```javascript
+"config:\n  ubuntu-core:\n    autopilot: true\n    timezone: Europe/Berlin\n    hostname: localhost.localdomain\n"
+```
+
+## /1.0/events
+
+Not implemented yet; poll on `.../operations/` until this is final.
+
+This URL isn't a real REST API endpoint, instead doing a GET query on it will
+upgrade the connection to a websocket on which notifications will be sent.
+
+### GET
+
+* Description: websocket upgrade
+* Authorization: trusted
+* Operation: sync
+* Return: none (never ending flow of events)
+
+#### Supported arguments
+* type: comma separated list of notifications to subscribe to (defaults to “operations,logging”)
+
+#### Notification types
+* operations
+* logging
+
+This never returns. Each notification is sent as a separate JSON dict:
+
+```javascript
+{
+   'timestamp': "1415639996457234",        # Current timestamp
+   'type': "operations",                   # Notification type
+   'operation': {...}                      # Operation object
+}
+
+{
+   'timestamp': "141563999675243",
+   'type': "logging",
+   'log': {'message': "Package [name] installed"}
+}
+```
+
+## /1.0/operations/<uuid>
+
+### GET
+
+* Description: background operation
+* Authorization: trusted
+* Operation: sync
+* Return: dict representing a background operation
+
+#### Sample result:
+
+```javascript
+{
+ "created_at": "1415639996123456",      // Creation timestamp
+ "may_cancel": true,
+ "output": {},
+ "resource": "/1.0/packages/camlistore.sergiusens",
+ "status": "running",                   // or “succeeded” or “failed”
+ "updated_at": "1415639996451214"       // Last update timestamp
+}
+```
+
+### DELETE
+
+All background operations will have `may_cancel` set to false.
```

Comment:
Why is that? Because we have not implemented "cancel" yet? Or am I misreading (again)?